src/vmdssd.c

```diff
@@ -63,20 +70,41 @@ static char *get_slot_from_syspath(char *path)
 
 static void get_ctrl(enum ibpi_pattern ibpi, uint16_t *new)
 {
-	switch (ibpi) {
-	case IBPI_PATTERN_LOCATE:
-		*new = ATTENTION_LOCATE;
-		break;
-	case IBPI_PATTERN_FAILED_DRIVE:
-		*new = ATTENTION_FAILURE;
-		break;
-	case IBPI_PATTERN_REBUILD:
-		*new = ATTENTION_REBUILD;
-		break;
-	default:
-		*new = ATTENTION_OFF;
-		break;
+	const struct ibpi_value *tmp = NULL;
+	int i = 0;
+
+	while (i < ARRAY_SIZE(ibpi_to_attention)) {
```

Comment:
please change it to the same as in NPEM, or event better- prepare functions for traveling through those ibpi_value maps